beater/api/mux.go

```diff
@@ -320,9 +320,13 @@ func backendRequestMetadata(c *request.Context) model.APMEvent {
 }
 
 func rumRequestMetadata(c *request.Context) model.APMEvent {
+	nat := new(model.NAT)
+	if c.SourceNATIP != nil {
+		nat = &model.NAT{IP: c.SourceNATIP, Port: c.SourceNATPort}
+	}
 	return model.APMEvent{
 		Client:    model.Client{IP: c.ClientIP},
-		Source:    model.Source{IP: c.SourceIP, Port: c.SourcePort},
+		Source:    model.Source{IP: c.SourceIP, Port: c.SourcePort, NAT: nat},
```

Comment:
Can we change this slightly to avoid allocating when `c.SourceNATIP == nil`? 

We can do so by creating the event first and then setting the `event.Source.NAT`.

```go
event := model.APMEvent{
 	Client:    model.Client{IP: c.ClientIP},
 	Source:    model.Source{IP: c.SourceIP, Port: c.SourcePort},
 	Timestamp: c.Timestamp,
 	UserAgent: model.UserAgent{Original: c.UserAgent,
}
if c.SourceNATIP != nil {
 	event.Source.NAT = &model.NAT{IP: c.SourceNATIP, Port: c.SourceNATPort}
}
return event
```